### ISSUE TITLE
Add Algorithms for Qiskit

### ISSUE THREAD

**Issue by born-2learn** [QPower-Research/QPowerAlgo#11]
- [x] Grover's Search
- [ ] Quantum Teleportation
- [ ] Bernstein- Vazirani
- [x] Shor's Algorithm
- [ ] Deutsch-Jozsa Algorithm
- [ ] QPE(Quantum Phase Estimation)
- [ ] VQE(Variational Quantum Eigensolver) for the specific case of Hydrogen
- [ ] QKD(Quantum Key Distribution)
- [x] QAOA(Quantum Approximate Optimization Algorithm)
- [x] QSVM(Quantum Support Vector Machine)
- [ ] VQC(Variational Quantum Classifier)

**Comment by rawatraghav** [comment]
I can take up Bernstein- Vazirani algorithm.

**Comment by diamondpowell** [comment]
I'd like to take Quantum Teleportation and Deutsch-Jozsa Algorithm

**Comment by born-2learn** [comment]
@rawatraghav Sure, you can proceed. Thank you for your contributions.

**Comment by born-2learn** [comment]
@DiamondPurvis Sure, you can proceed. Thank you for your contributions.

**Comment by JillisaJeffery** [comment]
I'll take Shor's algorithms and Quantum Teleportation!

**Comment by born-2learn** [comment]
@JillisaJeffery You can take up Shor's Algorithm. Quantum Teleportation is already taken by @DiamondPurvis.

**Comment by JillisaJeffery** [comment]
> @JillisaJeffery You can take up Shor's Algorithm. Quantum Teleportation is already taken by @DiamondPurvis.

hello I am sorry, can I do Q#?

**Comment by ghost** [comment]
can you please assign this issue to me?

**Comment by aryashah2k** [comment]
> * [x]  Grover's Search
> * [ ]  Bernstein- Vazirani
> * [x]  Shor's Algorithm
> * [ ]  Deutsch-Jozsa Algorithm
> * [ ]  QDK
> * [ ]  Quantum Teleportation

@born-2learn @DiamondPurvis @rawatraghav , Can I suggest addition of QAOA(Quantum Approximate Optimization Algorithm), Quantum Support Vector Machines(QSVM)- I will be able to contribute with these two as well.
Also, I would love taking up Bernstein Vazirani or Deutsch Josza Algorithm infact any from the existing list except QDK(I'm not aware of that) as my part of contribution .

**Comment by born-2learn** [comment]
@aryashah2k You can get started with QAOA and QSVM for qiskit. Bernstein Vazirani has been taken up by @rawatraghav and Deutsch-Jozsa has been taken up by @DiamondPurvis, let's wait and see if they are still interested in contributing else I can assign it to you. And yes, there was a typo in the list: it's QKD(Quantum Key Distribution) instead of QDK.

**Comment by born-2learn** [comment]
@Ak-panchal which algorithm would you like to work on specifically?

**Comment by Ak-panchal** [comment]
@born-2learn  can I work on shor's

**Comment by aryashah2k** [comment]
> @aryashah2k You can get started with QAOA and QSVM for qiskit. Bernstein Vazirani has been taken up by @rawatraghav and Deutsch-Jozsa has been taken up by @DiamondPurvis, let's wait and see if they are still interested in contributing else I can assign it to you. And yes, there was a typo in the list: it's QKD(Quantum Key Distribution) instead of QDK.

Yes great! I would love working on those two and am willing to collaborate with @rawatraghav or @DiamondPurvis if need be.
Alright, Thanks! @born-2learn , Starting this at the earliest, Also, where can I join for logistical inquiries and doubts like creating pull request, merge etc since I am relatively new to open source contributions...?

**Comment by ghost** [comment]
@born-2learn  Shor's or  Deutsch-Jozsa Algorithm

**Comment by born-2learn** [comment]
You can proceed with Shor's Algo for now, @Ak-panchal

**Comment by rashmitha520** [comment]
can I work on quantum teleportation as part of dwoc?

**Comment by aryashah2k** [comment]
Hello there @born-2learn , Last 14 days of DWOC are left and I would like to work on any pending issues under qiskit, If there is any way I can help with any issues please let me know, looking forward to contribute to the same!
Also I have a pending PR waiting to be merged #35 and have created fresh issues for QFT #36 , Im also working on Random Walks #34 and might open a PR by tomorrow...please look into these at the earliest
Regards,
Arya.
